.travis.yml

```diff
@@ -1,45 +1,56 @@
 # Travis CI Configuration File
 
+# Tell Travis CI which distro to use
+dist: trusty
+
 # Tell Travis CI we're using PHP
 language: php
 
-# Tell Travis CI which distro to use
-dist: trusty
+# Run Matrix for these PHP versions
+php:
+- "5.6"
+- "7.0"
+- "7.2"
 
-# Setup a global environemnt and overide as needed
 env:
+  # Global variable is re-defined in matrix-include -list
   global:
-    - WP_TRAVISCI=phpunit
+  - WP_TRAVISCI=phpunit
+  # Run phpunit in Travis matrix for these combinations
+  matrix:
+  - WP_MODE=single WP_BRANCH=master
+  - WP_MODE=single WP_BRANCH=latest
+  - WP_MODE=single WP_BRANCH=previous
+  - WP_MODE=multi  WP_BRANCH=master
 
-cache:
-  directories:
-   - $HOME/.composer/cache/files
-   - $HOME/.cache/yarn
-   - $HOME/.phpbrew
-
-# Next we define our matrix of additional build configurations to test against.
+# Define a matrix of additional build configurations
 # The versions listed above will automatically create our first configuration,
 # so it doesn't need to be re-defined below.
-
-# Test WP trunk/master and two latest versions on minimum (5.2).
-# Test WP latest two versions (4.5, 4.3) on most popular (5.6, 7.0).
-# Test WP trunk/master on edge platforms (PHP 7.2).
-
-# WP_VERSION specifies the tag to use. The way these tests are configured to run
-# requires at least WordPress 3.8. Specify "master" to test against SVN trunk.
-
 matrix:
   include:
+  # Versions listed below run within Precise distribution with PHP 5.2
```

Comment:
Perhaps we could elaborate here that only `precise` (and not `trusty`) dist has PHP 5.2 available?

I.e. comment on _why_, not only _what_

Reply:
I've updated the comment to include this, thanks!